src/Http/Routing/src/Matching/DfaMatcher.cs

```diff
@@ -16,13 +16,15 @@ internal sealed class DfaMatcher : Matcher
         private readonly EndpointSelector _selector;
         private readonly DfaState[] _states;
         private readonly int _maxSegmentCount;
+        private readonly bool _isDefaultEndpointSelector;
 
         public DfaMatcher(ILogger<DfaMatcher> logger, EndpointSelector selector, DfaState[] states, int maxSegmentCount)
         {
             _logger = logger;
             _selector = selector;
             _states = states;
             _maxSegmentCount = maxSegmentCount;
+            _isDefaultEndpointSelector = selector is DefaultEndpointSelector;
```

Comment:
I'd feel slightly better if this was an exact type equality check. This isn't public, but that would capture the intent better.

Reply:
Isn’t it sealed? Future proofing?